docs/core-concepts.md

```diff
@@ -0,0 +1,148 @@
+From within a viewModel, you call `setState { copy(yourProp = newValue) }`. If this syntax is unfamiliar:
+1. The signature of the lambda is `S.() -> S` meaning the receiver (aka `this`) of the lambda is the current state when the lambda is invoked and it returns the new lambda
+1. `copy` comes from the fact that the state class is a Kotlin [data class](https://kotlinlang.org/docs/reference/data-classes.html)
+1. The lambda is _not_ executed synchronously. It is put on a queue and run on a background thread. See [threading](threading.md) for more information
+
+#### Subscribing to state changes
+You can subscribe to state changes in your ViewModel. You may want to do this for analytics, for example. This usually done in the `init { ... }` block.
+
+```kotlin
+onEach { state ->
+}
+```
+
+```kotlin
+onEach(YourState::propA) { a ->
+}
+onEach(YourState::propA, YourState::propB, YourState::propC) { a, b, c ->
+}
+```
+If you are calling `setState` from within an `onEach` block, consider using a derived property.
+
+#### stateFlow
+`MavericksViewModel` exposes a `stateFlow` which is a normal Kotlin Flow and can be used however you would like. Helpers such as `onEach` above are just wrappers around it with automatic lifecycle cancellation.
+
+#### Accessing state once
+If you just want to retrieve the value of state one time, you can use `withState { state -> ... }`.
+
+When called from within a ViewModel, this will _not_ be run synchronously. It will be placed on a background queue so that all pending `setState` reducers are called prior to your `withState` call.
+
+# MavericksView
+`MavericksView` is where you actually render your state class to the screen. Most of the time, this will be a Fragment but it doesn't have to be.
+By implementing `MavericksView`, you:
+1. You can get access to a `MavericksViewModel` via any of the view model delegates. Doing so will automatically subscribe to changes and call `invalidate()`.
+1. Override `invalidate()`. It will get called any time the state for any view model accessed by the above delegates changes.
+
+#### ViewModel delegates
+1. `activityViewModel()` scopes the ViewModel to the Activity. All Fragments within the Activity that request a ViewModel of this type will receive the same instance. Useful for sharing data between screens.
+1. `fragmentViewModel()` scopes the ViewModel to the Fragment. It will be accessible to children fragments but parent or sibling fragments would get a different instance.
+1. `parentFragmentViewModel()` walks up the parent fragment tree until it finds one that has already created a ViewModel of the desired type. If none is found, it will automatically create one scoped to the highest parent fragment.
+1. `existingViewModel()` Same as `activityViewModel()` except it throws if the ViewModel was not already created by somebody else.
+
+If you want multiple ViewModels of the same type, you can pass a `keyFactory` into any of the delegates.
+
+#### Subscribing to state manually
```

Comment:
this seems to be covered in the `Subscribing to state changes` section above

Reply:
That is in the ViewModel section, this is the View section